mycroft/skills/skill_loader.py

```diff
@@ -243,13 +244,52 @@ def _get_last_modified_time(path):
     # Ensure modification times are valid
     bad_times = _bad_mod_times(mod_times)
     if bad_times:
-        raise OSError('{} had bad modification times'.format(bad_times))
+        raise OSError(f'{bad_times} had bad modification times')
     if all_files:
         return max(os.path.getmtime(f) for f in all_files)
     else:
         return 0
 
 
+def get_skill_class(skill_module):
+    """Find MycroftSkill based class in skill module.
+
+    Arguments:
+        skill_module (module): module to search for Skill class
+
+    Returns:
+        (MycroftSkill): Found subclass of MycroftSkill or None.
+    """
+    candidates = []
+    for name, obj in skill_module.__dict__.items():
+        if isclass(obj):
+            if issubclass(obj, MycroftSkill) and obj is not MycroftSkill:
+                candidates.append(obj)
+
+    if len(candidates) > 1:
+        LOG.warning(f"Multiple skills found in a single file!\n"
```

Comment:
Per above, I don't think this is necessarily true

Reply:
It will be after the code snippet you proposed below ;)